### DISCUSSION TITLE
X-modoboa-id:-Header

### DISCUSSION THREAD

**Discussion by patrickbenkoetter** [modoboa/modoboa#532] [Ideas]
If a mailservice gets exploited and senders are allowed to use any envelope sender address, it becomes hard to identify messages stuck in the queue that stem from the exploit.

Modo should provide means to associate an unique hash with each account. The hash must allow to relate to any privacy sensitive account information.

Once the Tracking-ID is in place a policy service could add as X-Header it to every mail that is being accepted. When the message leaves the system via SMTP or LMTP header checks can remove it.

**Comment by tonioo** [comment]
What would you need, an API ?

**Comment by patrickbenkoetter** [comment]
I don't think so. We would need a service and the service would need to query the database for the logins Tracking-ID and add that as header to the message in process.

**Comment by tonioo** [comment]
Hi,
could you give an example postfix configuration for such a feature please?

Otherwise, is a mechanism like this one still relevant when sender_login_maps are in use ?
